### PR TITLE
Add two sidecar layouts to layout component

### PR DESCRIPTION
**Pull request by wibjorn**
Task: task-925514
Task: task-925513

Link: preview-586
Page specific link: https://design.learn.microsoft.com/pulls/586/components/layout.html

This pull request adds a third possible layout to the `.layout` component. See documentation page for details. Also fixes a mobile navigation issue introduced late in the review process for #582 

## Testing

1. Visit documentation page.
2. Select "Show container labels" button
3. Select "Sidecar left" and "Sidecar right" from the layout buttons.
4. Read through code.

## Contributor checklist

- [x] Did you update the description of this pull request with a review link and test steps?
- [x] Did you submit a changeset? Changesets are required for all code changes.
- [ ] Does your pull request implement complex UI logic (js/ts)? Consider adding an integration test to test your user flow.
- [ ] Does your pull request add a new page to the documentation site? Add your new page for automated accessibility testing in /integration/tests/accessibility.

### CODE REVIEW

File: site/src/components/layout.md [hunk 1]
```diff
@@ -136,6 +140,76 @@ The specification for this layout is as follows.
...
+The "sidecar" remains as defined in the section above, but in `layout-sidecar-left` the sidecar refers to the `layout-body-aside` element, which sits to the right of `layout-body-main` on tablet screens and wider. Unlike other layouts, this layout does not allow the usage of the `layout-body-menu` containers.
+
+How do I apply it? `layout-sidecar-right` to the `layout` element.
+Required elements: all except `layout-body-hero` and `layout-body-menu` (see allowed elements).
+Allowed elements: all except `layout-body-menu`.
+
+The following block is arranged from narrow widths on the left to wider widths on the right.
+
+```txt
+ ┌──────────────┐ ┌──────────────────────┐ ┌──────────────────────┐
```
Comment (disimmon) [inline site/src/components/layout.md:188]: Below it says `Narrow all elements are stacked` but this layout diagram shows side x side, on narrow does main stack atop aside?

File: site/src/scaffold/styles/layout-buttons.scss [hunk 2]
```diff
@@ -0,0 +1,10 @@
+$layouts: 'single', 'holy-grail', 'sidecar-left', 'sidecar-right';
+
+@each $layout in $layouts {
+	.layout-#{$layout} {
+		[data-set-layout='layout-#{$layout}'] {
+			@extend .button-primary;
+			@extend .button-filled;
+		}
+	}
+}
```
Comment (BenDMyers) [inline site/src/scaffold/styles/layout-buttons.scss:10]: Bit of a simpler approach, given we apply `aria-pressed` and can use it as a [stateful, semantic selector](https://benmyers.dev/blog/semantic-selectors/)?

```suggestion
[data-set-layout][aria-pressed="true"] {
		@extend .button-primary;
		@extend .button-filled;
}
```

Comment (wibjorn) [reply site/src/scaffold/styles/layout-buttons.scss:10]: Looks good to me, a lot less output.

File: site/src/scaffold/styles/layout-buttons.scss [hunk 3]
```diff
@@ -0,0 +1,6 @@
+$layouts: 'single', 'holy-grail', 'sidecar-left', 'sidecar-right';
+
```
Comment (BenDMyers) [inline site/src/scaffold/styles/layout-buttons.scss:2]: Seems this line can be removed now, too:

```suggestion
```

Comment (wibjorn) [reply site/src/scaffold/styles/layout-buttons.scss]: Good shout! Removed

File: site/src/components/layout.md (hunk 1 above)
Comment (wibjorn) [reply site/src/components/layout.md:188]: Fixed

**Review by disimmon — DISMISSED** [review]

File: site/src/components/layout.md [hunk 4]
```diff
@@ -136,6 +140,76 @@ The specification for this layout is as follows.
...
+
+A "sidecar" is a smaller companion container that sits beside the main content container on tablet screens and wider. In `layout-sidecar-left` the sidecar refers to the `layout-body-menu` element, which sits to the left of `layout-body-main`. Unlike other layouts, this layout does not allow the usage of the `layout-body-aside` containers.
+
+How do I apply it? `layout-sidecar-left` to the `layout` element.
+Required elements: all except `layout-body-hero` and `layout-body-aside` (see allowed elements).
+Allowed elements: all except `layout-body-aside`.
+
+The following block is arranged from narrow widths on the left to wider widths on the right.
+
+```txt
```
Comment (homing1) [inline site/src/components/layout.md:153]: Would it be helpful to label these blocks to supplement the previous sentence? Something like this?

image

File: site/src/components/layout.md [hunk 5]
```diff
@@ -136,6 +140,76 @@ The specification for this layout is as follows.
 | Desktop    | Menu, main, aside are side by side. Main is wider than menu.                                                                  |
 | Widescreen | Same as desktop with scaling gutter that keeps combined width of menu, main, aside to the width of the widescreen breakpoint. |
 
+### Sidecar left layout
+
+A "sidecar" is a smaller companion container that sits beside the main content container on tablet screens and wider. In `layout-sidecar-left` the sidecar refers to the `layout-body-menu` element, which sits to the left of `layout-body-main`. Unlike other layouts, this layout does not allow the usage of the `layout-body-aside` containers.
+
+How do I apply it? `layout-sidecar-left` to the `layout` element.
```
Comment (homing1) [inline site/src/components/layout.md:147]: Can we add a new line between these three sentences like in the single layout? It seems hard to parse the required elements and allowed elements, especially for sidecar-left/right

image

Comment (wibjorn) [reply site/src/components/layout.md:147]: Weird, guess the single newline is collapsed. I'll update all the ones after single.

File: site/src/components/layout.md (hunk 4 above)
Comment (wibjorn) [reply site/src/components/layout.md:159]: Nothing wrong with some better labelling. Updated all these text blocks on this page.

File: site/src/components/layout.md (hunk 5 above)
Comment (wibjorn) [reply site/src/components/layout.md:151]: Updated

**Review by homing1 — APPROVED** [review]
